From 295a4a7586cc1f4f867a92c864cc1c49eab0991c Mon Sep 17 00:00:00 2001
From: Jiri Slaby <jslaby@suse.cz>
Date: Mon, 4 Nov 2019 10:00:37 +0100
Subject: [PATCH] CMakeLists: make static lib optional

Build static lib only if BUILD_STATIC_LIBS is specified.
---
 CMakeLists.txt | 29 ++++++++++++++++++-----------
 1 file changed, 18 insertions(+), 11 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6e00ca0..5e493de 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,6 +6,7 @@ project(minisat)
 # Configurable options:
 
 option(STATIC_BINARIES "Link binaries statically." ON)
+option(BUILD_STATIC_LIBS "Build static library." ON)
 option(USE_SORELEASE   "Use SORELEASE in shared library filename." ON)
 
 #--------------------------------------------------------------------------------------------------
@@ -51,11 +52,19 @@ set(MINISAT_LIB_SOURCES
     minisat/core/Solver.cc
     minisat/simp/SimpSolver.cc)
 
-add_library(minisat-lib-static STATIC ${MINISAT_LIB_SOURCES})
-add_library(minisat-lib-shared SHARED ${MINISAT_LIB_SOURCES})
+if (BUILD_STATIC_LIBS OR STATIC_BINARIES)
+	add_library(minisat-lib-static STATIC ${MINISAT_LIB_SOURCES})
+	target_link_libraries(minisat-lib-static ${ZLIB_LIBRARY})
+	set_target_properties(minisat-lib-static PROPERTIES OUTPUT_NAME "minisat")
+endif()
 
+add_library(minisat-lib-shared SHARED ${MINISAT_LIB_SOURCES})
 target_link_libraries(minisat-lib-shared ${ZLIB_LIBRARY})
-target_link_libraries(minisat-lib-static ${ZLIB_LIBRARY})
+set_target_properties(minisat-lib-shared
+  PROPERTIES
+    OUTPUT_NAME "minisat"
+    VERSION ${MINISAT_VERSION}
+    SOVERSION ${MINISAT_SOVERSION})
 
 add_executable(minisat_core minisat/core/Main.cc)
 add_executable(minisat_simp minisat/simp/Main.cc)
@@ -68,13 +77,6 @@ else()
   target_link_libraries(minisat_simp minisat-lib-shared)
 endif()
 
-set_target_properties(minisat-lib-static PROPERTIES OUTPUT_NAME "minisat")
-set_target_properties(minisat-lib-shared
-  PROPERTIES
-    OUTPUT_NAME "minisat" 
-    VERSION ${MINISAT_VERSION}
-    SOVERSION ${MINISAT_SOVERSION})
-
 set_target_properties(minisat_simp       PROPERTIES OUTPUT_NAME "minisat")
 
 #--------------------------------------------------------------------------------------------------
@@ -82,7 +84,12 @@ set_target_properties(minisat_simp       PROPERTIES OUTPUT_NAME "minisat")
 
 set(CMAKE_INSTALL_LIBDIR lib CACHE STRING "Output directory for libraries")
 
-install(TARGETS minisat-lib-static minisat-lib-shared minisat_core minisat_simp 
+if (BUILD_STATIC_LIBS)
+	install(TARGETS minisat-lib-static
+		ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+endif()
+
+install(TARGETS minisat-lib-shared minisat_core minisat_simp
         RUNTIME DESTINATION bin
         LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
         ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
-- 
2.23.0

